checkov/sca_package/scanner.py

```diff
@@ -2,103 +2,110 @@
-
-    def scan(self, input_output_paths: "Iterable[Tuple[Path, Path]]", cleanup_twictcli: bool = True) \
-            -> "Sequence[Dict[str, Any]]":
-        self.setup_twictcli()
+        self.base_url = bc_integration.api_url
+        self.headers = merge_dicts(
+            get_default_get_headers(bc_integration.bc_source, bc_integration.bc_source_version),
+            {"Authorization": bc_integration.get_auth_token()},
+        )
 
+    def scan(self, input_paths: "Iterable[Path]") \
+            -> "Sequence[Dict[str, Any]]":
         scan_results = asyncio.run(
-            self.run_scan_multi(
-                address=package_scanning_integration.get_proxy_address(),
-                bc_api_key=package_scanning_integration.get_bc_api_key(),
-                input_output_paths=input_output_paths,
-            )
+            self.run_scan_multi(input_paths=input_paths)
         )
-        if cleanup_twictcli:
-            self.cleanup_twictcli()
         return scan_results
 
     async def run_scan_multi(
-        self,
-        address: str,
-        bc_api_key: str,
-        input_output_paths: "Iterable[Tuple[Path, Path]]",
+            self,
+            input_paths: "Iterable[Path]",
     ) -> "Sequence[Dict[str, Any]]":
-        args = [
-            (
-                f"./{TWISTCLI_FILE_NAME} coderepo scan --address {address} --token {bc_api_key} --output-file '{output_path.absolute()}' '{input_path.absolute()}'",
-                input_path,
-                output_path,
-            )
-            for input_path, output_path in input_output_paths
-        ]
         if os.getenv("PYCHARM_HOSTED") == "1":
             # PYCHARM_HOSTED env variable equals 1 when running via Pycharm.
             # it avoids us from crashing, which happens when using multiprocessing via Pycharm's debug-mode
             logging.warning("Running the scans in sequence for avoiding crashing when running via Pycharm")
             scan_results = []
-            for curr_arg in args:
-                scan_results.append(await self.run_scan(*curr_arg))
+            for input_path in input_paths:
+                scan_results.append(self.run_scan(input_path))
         else:
-            async with Pool() as pool:
-                scan_results = await pool.starmap(self.run_scan, args)
+            input_paths = [(input_path,) for input_path in input_paths]
+            with Pool() as pool:
+                scan_results = pool.starmap(self.run_scan, input_paths)
 
         return scan_results
 
-    async def run_scan(
-        self,
-        command: str,
-        input_path: Path,
-        output_path: Path,
-    ) -> Dict[str, Any]:
+    def run_scan(self, input_path: Path) -> dict:
         logging.info(f"Start to scan package file {input_path}")
-        process = await asyncio.create_subprocess_shell(
-            command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+
+        headers = merge_dicts(
```

Comment:
maybe create a method "get_header" in this class. we have the same logic in line 22.
if it isn't to hard, the best is create some "global" function "get_header" (which can be part of 'bc_integration') and than we can use also it instead the header-creation in image_scanner.py:118 (and apparently in a lot of other places), but it may be more complicated, so let's start do it in this class